src/Symfony/Component/Notifier/Notifier.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Notifier\Channel\ChannelInterface;
+use Symfony\Component\Notifier\Channel\ChannelPolicy;
+use Symfony\Component\Notifier\Channel\ChannelPolicyInterface;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Notification\Notification;
+use Symfony\Component\Notifier\Receiver\AdminReceiver;
+use Symfony\Component\Notifier\Receiver\Receiver;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class Notifier implements NotifierInterface
+{
+    private static $adminReceivers = [];
+
+    private $channels;
+    private $policy;
+    private $bus;
+
+    /**
+     * @param ChannelInterface[]|ContainerInterface $channels
+     */
+    public function __construct($channels, ChannelPolicyInterface $policy = null, MessageBusInterface $bus = null)
+    {
+        $this->channels = $channels;
+        $this->policy = $policy;
+        $this->bus = $bus;
+    }
+
+    public function send(Notification $notification, Receiver ...$receivers): void
+    {
+        foreach ($receivers as $receiver) {
+            foreach ($this->getChannels($notification, $receiver) as $channel => $transport) {
+                $channel->notify($notification, $receiver, $transport, $this->bus);
+            }
+        }
+    }
+
+    public static function addAdminReceiver(AdminReceiver $receiver): void
+    {
+        self::$adminReceivers[] = $receiver;
+    }
+
+    /**
+     * @return AdminReceiver[]
+     */
+    public static function getAdminReceivers(): array
```

Comment:
What is the purpose of making these two methods (`addAdminReceiver` & `getAdminReceivers`) static ? As far as I can see, all the places where they are used have the service instance `notifier`.

Reply:
That was needed in another impl, but that's not a requirement anymore, code updated.